docs/development/pigs/pig-006.md

```diff
@@ -0,0 +1,116 @@
+
+# PIG 6 - CTA observation handling in Gammapy
+
+* Author: David Fidalgo, Christoph Deil, Régis Terrier et al
+* Created: October 10, 2018
+* Accepted: --
+* Status: Being discussed
+* Discussion: --
+
+## Abstract
+
+In this PIG we want to outline an improvement of the CTA observation handling in Gammapy.
+The current handling has some limitations and misses some key features, like selecting a given time interval and performing an analysis on it.
+We also want to start relying more on GTIs instead  
+Since specifics of the final data format of CTA are still unclear, this PIG does not aim for a final implementation of CTA observation handling, nor the handling of all types of data from various instruments forseen for Gammapy in the future.
+It focuses on trying to resolve current limitations when analyzing IACT data.
+This PIG emerged at the Coding Spring in Madrid 2018.
+
+## Some terminology
+
+This terminology and the definition of the terms will probably change or be expanded once the final data format of CTA is decided.
+For now we will use following definitions to go forward in Gammapy:
+
+* **GTI (good time interval)**:
+  * time intervals for which the instrument was on
+  * the sum over these time intervals equals the *on time*
+
+
+* **Observation**:
+  * has a unique identifier, the *obs_id*
+  * contains 1 event list, 1 GTI table and 1 corresponding set of IRFs (aeff, edisp, psf)
```

Comment:
note that the IRFs are associated to a GTI, not to the observation in general. E.g. you might have a single GTI, and then it's as we have now, or you might have 2 or more GTIs in the observation, where the calibration is quite different (e.g. a GTI with perfect conditions, and several others others with cloudy but still usable conditions).

Reply:
@kosack - Both Gammapy and ctools currently assume one IRF per observation.

To quote from http://cta.irap.omp.eu/gammalib/users/user_manual/modules/obs.html

> A gamma-ray observation is defined for a single specific instrument, and describes a time period during which the instrument is in a given stable configuration that can be characterized by a single specific response function. Each gamma-ray observation is composed of events and a response function.

If you want to distribute one IRF per GTI, we can implement that in Gammapy in the coming months, but it's a big decision, not what was assumed from the science tools so far.

That said, we'll probably anyways have to introduce a major change to support EVENT_TYPE, where you have one IRF per OBS_ID and EVENT_TYPE.

One option on the Gammapy side could be to stick with one IRF per OBS_ID for now and just to introduce the EVENT and GTI filter functionality, work with you and CTA to define the new DL3 model, and once there is a concrete proposal, prototype that early next year.